src/internal/cache-client.ts

```diff
@@ -208,36 +210,35 @@ export class CacheClient {
         },
         (err, resp) => {
           if (resp) {
-            const momentoResult = momentoResultConverter(resp.result);
-            if (
-              momentoResult !== CacheGetStatus.Miss &&
-              momentoResult !== CacheGetStatus.Hit
-            ) {
-              reject(new UnknownServiceError(resp.message));
+            switch (resp.result) {
+              case cache.cache_client.ECacheResult.Miss:
+                resolve(new CacheGet.Miss());
+                break;
+              case cache.cache_client.ECacheResult.Hit:
+                resolve(new CacheGet.Hit(resp.cache_body));
+                break;
+              case cache.cache_client.ECacheResult.Invalid:
+              case cache.cache_client.ECacheResult.Ok:
+                resolve(new CacheGet.Error(new UnknownError(resp.message)));
```

Comment:
this reads a little weird to me. If the result is `OK` we return an error? I am assuming this is because get should only return a hit or a miss?

Reply:
yeah the wire format is kind of weird on this one, but i think it would be a bug for the server to return anything but Hit here.  this is a little different from the .net sdk though, which does `if Miss ... else return Hit`.

this one feels probably more correct to me though.